modules/fbx/extensions/fbx_document_extension_convert_importer_mesh.h

```diff
@@ -0,0 +1,47 @@
+/**************************************************************************/
+/*  fbx_document_extension_convert_importer_mesh.h                        */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef FBX_DOCUMENT_EXTENSION_CONVERT_IMPORTER_MESH_H
+#define FBX_DOCUMENT_EXTENSION_CONVERT_IMPORTER_MESH_H
+
+#include "modules/gltf/extensions/gltf_document_extension.h"
+#include "modules/gltf/gltf_defines.h"
+
+class FBXDocumentExtensionConvertImporterMesh : public GLTFDocumentExtension {
```

Comment:
Since FBXDocumentExtensionConvertImporterMesh extends GLTFDocumentExtension, what are your thoughts on using GLTFDocumentExtensionConvertImporterMesh directly?

Reply:
Not a fan of code duplication.

Reply:
> Not a fan of code duplication.

yea im also Not a fan of code duplication but sometimes code readability is more important

Reply:
> Since FBXDocumentExtensionConvertImporterMesh extends GLTFDocumentExtension, what are your thoughts on using GLTFDocumentExtensionConvertImporterMesh directly?


sounds right but i think this might make things abit too spaghetti-like for anyone who wants to maintain it or extend it
if we want to remove this new class and reuse the gltf one we should comment that somewhere since this fbx importer is already confusing enough

since now fbx also supports extensions shouldn't we indicate that somewhere to other developers?
currently with this class most people can tell just by looking at the docs that there is both gltf and fbx improter extensions which inherit from gltf document extension
also it makes it easier to extend it later too and add more functionality to it

Reply:
Let me ask for a second opinion.